include/pybind11/cast.h

```diff
@@ -870,7 +910,7 @@ template <typename TypeCaster> TypeCaster &load_type(TypeCaster &conv, const han
         throw cast_error("Unable to cast Python instance to C++ type (compile in debug mode for details)");
 #else
         throw cast_error("Unable to cast Python instance of type " +
-            (std::string) handle.get_type().str() + " to C++ type '" + type_id<T>() + "''");
+            (std::string) handle.get_type().str() + " to C++ type '" + type_id<TypeCaster>() + "''");
```

Comment:
(temporary workaround to get pybind11 to compile in debug mode -- would be nicer to have the actual type here)

cc @jagerman